TFT/src/User/API/parseACK.c

```diff
@@ -538,789 +531,790 @@ void parseACK(void)
-    {                              // forward the message to the supplementary serial port
-      Serial_Puts(serialPort[ack_port_index].port, dmaL2Cache);
 
-      if (!infoHost.wait)         // if "ok" has been received
-        ack_port_index = PORT_1;  // reset ACK port index to avoid wrong relaying (in case no more commands will
-                                  // be sent by interfaceCmd) of any successive spontaneous ACK message
-    }
     #ifdef SERIAL_PORT_2
-      else if (!infoHost.wait && !ack_cmp("ok"))  // if the ACK message is not related to a gcode originated by the TFT and
-      {                                           // it is not "ok", it is a spontaneous ACK message
-        // pass on the spontaneous ACK message to all the supplementary serial ports (since these messages come unrequested)
-        for (SERIAL_PORT_INDEX i = PORT_2; i < SERIAL_PORT_COUNT; i++)
-        {
-          // forward data only if serial port is enabled
-          if (infoSettings.serial_port[i] > 0
-              #ifdef SERIAL_DEBUG_PORT
-                && serialPort[i].port != SERIAL_DEBUG_PORT  // do not forward data to serial debug port
-              #endif
-             )
+      if (ack_port_index == PORT_1)
+      {
+        if (infoHost.wait == false && !ack_starts_with("ok"))
+        { // if the ACK message is not related to a gcode originated by the TFT and
+          // it is not "ok", it is a spontaneous ACK message, so pass it to all the
+          // supplementary serial ports (since these messages came unrequested)
+          for (SERIAL_PORT_INDEX i = PORT_2; i < SERIAL_PORT_COUNT; i++)
           {
-            Serial_Puts(serialPort[i].port, dmaL2Cache);  // pass on the ACK message to the port
+            // forward data only if serial port is enabled
+            if (infoSettings.serial_port[i] > 0
+                #ifdef SERIAL_DEBUG_PORT
+                  && serialPort[i].port != SERIAL_DEBUG_PORT  // do not forward data to serial debug port
+                #endif
+                )
+              Serial_Puts(serialPort[i].port, dmaL2Cache);  // pass on the ACK message to the port
           }
         }
       }
+      else
+      { // if the ACK message is related to a gcode originated by a supplementary serial port,
+        // forward the message to the supplementary serial port
+        Serial_Puts(serialPort[ack_port_index].port, dmaL2Cache);
+
+        if (infoHost.wait == false)  // if "ok" has been received
+          ack_port_index = PORT_1;   // reset ACK port index to avoid wrong relaying (in case no more commands will
+                                     // be sent by interfaceCmd) of any successive spontaneous ACK message
+      }
     #endif
   }
 }
 
 #ifdef SERIAL_PORT_2
 
-void parseRcvGcode(void)
-{
-  for (SERIAL_PORT_INDEX i = PORT_2; i < SERIAL_PORT_COUNT; i++)  // scan all the supplementary serial ports
+  void parseRcvGcode(void)
```

Comment:
for the functions in .c no indentation is used with `#ifdef`. Only for inline code or data structures / function definition, indentation is used. If it was depending on me, no indentation was used at all. Something similar to Marlin was used.

Reply:
Good to know, thank you!